tests/src/test/scala/cats/tests/FoldableTests.scala

```diff
@@ -204,6 +204,28 @@ class FoldableTestsAdditional extends CatsSuite {
     // test laziness of foldM
     dangerous.foldM(0)((acc, a) => if (a < 2) Some(acc + a) else None) should === (None)
   }
+
+  test(".foldLeftM short-circuiting") {
+    val ns = Stream.continually(1)
+    val res = Foldable[Stream].foldLeftM[Either[Int, ?], Int, Int](ns, 0) { (sum, n) =>
+      if (sum >= 100000) Left(sum) else Right(sum + n)
+    }
+    assert(res == Left(100000))
+  }
+
+  test(".foldLeftM short-circuiting optimality") {
```

Comment:
Huh, I didn't notice that I was testing some other code 😃. I suppose changing the signature of

```scala
def uncons: Option[(A, Source[A])]
```

to

```scala
def uncons: Option[(A, Eval[Source[A]])]
```

and doing the necessary changes would make it pass.